session/session.go

```diff
@@ -224,3 +241,26 @@ func getDefaultTransport(endpointURL string) TransportHandler {
 
 	return transportHandler
 }
+
+func isTimeout(err error) bool {
+	if slErr, ok := err.(sl.Error); ok {
+		switch slErr.StatusCode {
+		case 408, 504, 599:
+			return true
+		}
+	}
+
+	if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+		return true
```

Comment:
@minsikl Yes, admittedly you would have to configure timeout and retry appropriately in the session based on what you are trying to do.

This could lead to us creating more helper methods to enable something like this:

```go
sess2 := sess.Timeout(900).Retries(2) // and use sess2
// or
services.GetProductOrderService(sess.Retries(0)).PlaceOrder(...)
// or
services.GetVirtualGuestService(sess.Retries(2)).CreateObject(...)
```